addons/sale_pdf_quote_builder/models/sale_order_line.py

```diff
@@ -37,3 +37,8 @@ def _compute_available_product_document_ids(self):
                         ('res_id', '=', line.product_template_id.id),
                 ('attached_on_sale', '=', 'inside')
             ], order='res_model, sequence').ids
+
+            # Ensure selected documents are still in the available documents
+            line.product_document_ids = line.product_document_ids.filtered(
+                lambda doc: doc in line.available_product_document_ids
+            )
```

Comment:
Nope, don't modify another field in a compute, very bad practice.
should probably be an onchange.

Reply:
Indeed :facepalm: